src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -8543,15 +8547,16 @@ void MacroAssembler::char_array_compress(Register src, Register dst, Register le
     jcc(Assembler::zero, post_alignment);
 
     // ~(~0 << len), where len is the # of remaining elements to process
-    movl(result, 0xFFFFFFFF);
-    shlxl(result, result, tmp5);
-    notl(result);
-    kmovdl(mask2, result);
+    movl(len, 0xFFFFFFFF);
+    shlxl(len, len, tmp5);
+    notl(len);
+    kmovdl(mask2, len);
+    movl(len, result);
 
     evmovdquw(tmp1Reg, mask2, Address(src, 0), /*merge*/ false, Assembler::AVX_512bit);
     evpcmpw(mask1, mask2, tmp1Reg, tmp2Reg, Assembler::le, /*signed*/ false, Assembler::AVX_512bit);
     ktestd(mask1, mask2);
-    jcc(Assembler::carryClear, return_zero);
+    jccb(Assembler::carryClear, copy_tail);
```

Comment:
```suggestion
    jcc(Assembler::carryClear, copy_tail);
```
(in this case the jump can potentially be long)